model/task/db.go

```diff
@@ -194,6 +194,42 @@ var (
 					},
 					"then": evergreen.TaskTimedOut,
 				},
+				// A task will run if it is activated and does not have any blocking deps
```

Comment:
it looks like calling `FindTasksByVersion` will execute this code path  to calculate the displayStatus when [this code is ran](https://github.com/evergreen-ci/evergreen/blob/main/model/task/task.go#L2770) . We may also be able to utilize this code path for the task.FindByIdExecution (called in the Task resolver) and potentially remove the updatedStatus calculation logic in taskResolver.Status

Reply:
Good catch just added this